Rubberduck.RibbonDispatcher/AbstractCOM/IClickedEvents.cs

```diff
@@ -0,0 +1,20 @@
+﻿////////////////////////////////////////////////////////////////////////////////////////////////////
+//                                Copyright (c) 2017 Pieter Geerkens                              //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+using System;
+using System.Runtime.InteropServices;
+
+using Rubberduck.RibbonDispatcher.EventHandlers;
+
+namespace Rubberduck.RibbonDispatcher.AbstractCOM {
+    /// <summary>TODO</summary>
+    [ComVisible(true)]
+    [CLSCompliant(true)]
+    [InterfaceType(ComInterfaceType.InterfaceIsIDispatch)]
+    [Guid(RubberduckGuid.IClickedEvents)]
+    public interface IClickedEvents {
+        /// <summary>TODO</summary>
+        [DispId(1)]
```

Comment:
the other interfaces refer to `DispIds`. Why doesn't this one?

Reply:
I'm not sure if we should enforce .NET conventions of handling events across to the COM. In COM world, it's very rare to see an event return an args object; they are normally returned in a flat list of parameters. For those who only do VBA, it could come off as counter-intuitive. This approach also reduces the number of objects that must be exposed to COM; primitives poses less of a problem when you cross the boundary between two worlds.

Reply:
@Vogel612 L Laziness? The initial introduction of the DispIds enum revolved around having a single DispId value, across multiple interfaces, for each callback. These other interfaces stand on their own, and don't directly support the callbacks,, so I didn't see it as necessary.

Reply:
@Vogel612 : What are your thoughts on @bclothier 's suggestions? I was thinking on the same lines as I wrote the code.

Reply:
Frankly: I don't care. I'm not a user of Rubberduck, nor do I actually intend to be one, because that would mean I'm writing vba. Rubberduck is a great tool, but I really don't want to write VBA. As such I don't really go for one way or the other. Assuming that COM **usually** does things the way @bclothier outlined (which I cannot confirm), I'm inclined to agree here. What about you, @rubberduck-vba/dev ?

Reply:
@bclothier is entirely correct, @Vogel612: client code (VBA) will expect even parameters to be passed as a list of parameters; the .net-idiomatic `(object sender, EventArgs e)` signature would be extremely surprising in a VB6/VBA API.

...which kinda sucks, because this means that adhering to VB6/VBA/COM expectations means any changes to the interface (e.g. adding a parameter) breaks the API, whereas the .net way wouldn't break any client event handler.

Regardless, I think "the customer is always right", and so COM API's should adhere to what the clients are expecting.

Reply:
@retailcoder : Got it - I will make the changes.

Reply:
@retailcoder FWIW - a new member need not break the interfaces in COM's world, as long we adhere to the rule of never editing or re-arranging already used-up `DispID`. In COM's world, a version 1.1 of an interface is backward-compatible with client compiled using version 1.0 if all new members are added with higher `DispID`s. That is not true in the .NET, so as long you adhere to the convention of having .NET *never* using COM interfaces within its codebase, it should be fine. If it's necessary for some reason for both .NET and COM clients to subscribe to the events, then the implementation of COM event interface can just convert back to .NET's idiom and all .NET code can just use the direct .NET while COM implementation acts as a facade for COM compatibility. That way, any changes would be confined to at least 3 places -- the .NET class for the args, the COM implementation of the same args class, and the COM interface. Not ideal as having only one place to change but less sucky at least.

Reply:
I know. That's how and why `Excel.Range` has that lovely `.Value2` property.